library/goptions.mli

```diff
@@ -129,9 +139,16 @@ val declare_string_option: ?preprocess:(string -> string) ->
 val declare_stringopt_option: ?preprocess:(string option -> string option) ->
                               string option option_sig -> unit
 
-(** Helper to declare a reference controlled by an option. Read-only
+(** Helpers to declare a reference controlled by an option. Read-only
    as to avoid races. *)
+val declare_int_option_and_ref : depr:bool -> key:option_name -> value:int -> (unit -> int)
+val declare_intopt_option_and_ref : depr:bool -> key:option_name -> (unit -> int option)
+val declare_nat_option_and_ref : depr:bool -> key:option_name -> value:int -> (unit -> int)
 val declare_bool_option_and_ref : depr:bool -> key:option_name -> value:bool -> (unit -> bool)
+val declare_string_option_and_ref : depr:bool -> key:option_name -> value:string -> (unit -> string)
```

Comment:
Types seem common enough as to provide a `type 'a opt_decl = depr:bool -> key -> ...` ?